Bottom: ffbd44d04a08775494b692404a5f60f79bcf2fe6
Top:    eef5573a6a6d35813080bfe8a07a529cb96e5145
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-07-20 16:15:23 +0800

tuntap: fix

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 4dc61fd..3ae6a02 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -2436,12 +2436,6 @@ static int tun_xdp_one(struct tun_struct *tun,
 		goto out;
 	}
 
-	if (skb_xdp) {
-		err = do_xdp_generic(xdp_prog, skb);
-		if (err != XDP_PASS)
-			goto out;
-	}
-
 	skb_reserve(skb, xdp->data - xdp->data_hard_start);
 	skb_put(skb, xdp->data_end - xdp->data);
 
@@ -2456,6 +2450,12 @@ static int tun_xdp_one(struct tun_struct *tun,
 	skb_reset_network_header(skb);
 	skb_probe_transport_header(skb, 0);
 
+	if (skb_xdp) {
+		err = do_xdp_generic(xdp_prog, skb);
+		if (err != XDP_PASS)
+			goto out;
+	}
+
 	if (!rcu_dereference(tun->steering_prog))
 		rxhash = __skb_get_hash_symmetric(skb);
